### PR TITLE
Simple Service to make a GraphViz graph of the modules runtime dependencies

### PR DESCRIPTION
**Pull request by fwyzard**
Simple Service to make a GraphViz graph of the modules runtime dependencies:
  - draw hard dependencies according to the "consumes" dependencies;
  - draw soft dependencies to reflect the order of scheduled modue in each path;
  - draw SubProcesses in subgraphs.

Use GraphViz dot to generate an SVG representation of the dependencies:
```
dot -v -Tsvg dependency.gv -o dependency.svg
```

### CODE REVIEW

**Comment by fwyzard** [comment]
please test

File: FWCore/Services/plugins/BuildFile.xml [hunk 1]
```diff
@@ -1,6 +1,8 @@
 <use   name="FWCore/Services"/>
 <use   name="Utilities/StorageFactory"/>
+<use   name="boost"/>
 <use   name="gcc-atomic"/>
 <library   file="*.cc" name="FWCoreServicesPlugins">
   <flags   EDM_PLUGIN="1"/>
+  <flags   CXXFLAGS="-Wno-maybe-uninitialized"/>
```
Comment (Dr15Jones) [inline FWCore/Services/plugins/BuildFile.xml]: @smuzaffar are you OK with this flag?

Comment (fwyzard) [inline FWCore/Services/plugins/BuildFile.xml]: if there is a way f disabling the check only in the boost headers, that works for me as well...

Comment (smuzaffar) [inline FWCore/Services/plugins/BuildFile.xml]: @fwyzard , no there is no way to disable it only for boost. If you are worrying about getting these warnings in IB logs then we do ignore warnings coming from externals.

Comment (fwyzard) [inline FWCore/Services/plugins/BuildFile.xml]: The problem is that we have `-Werror=maybe-uninitialized` so they become actual build errors.
 If you prefer I can use `-Wno-error=maybe-uninitialized` to turn them back into warnings.

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
+1

**Comment by fwyzard** [comment]
OK, I have updated the service based on the feedback at the core software meeting:
  - moved it to a different plugin, so the warnings can be disabled only for this service
  - gather information about the Source module explicitly

**Comment by fwyzard** [comment]
please test

File: FWCore/Services/plugins/BuildFile.xml [hunk 2]
```diff
@@ -1,6 +1,12 @@
...
+<use   name="boost"/>
 <use   name="gcc-atomic"/>
 <library   file="*.cc" name="FWCoreServicesPlugins">
+  <flags   SKIP_FILES="DependencyGraph.cc"/>
   <flags   EDM_PLUGIN="1"/>
 </library>
+<library   file="DependencyGraph.cc" name="FWCoreServicesDependencyGraphPlugin">
+  <flags   EDM_PLUGIN="1"/>
+  <flags   CXXFLAGS="-Wno-maybe-uninitialized"/>
+</library>
```
Comment (smuzaffar) [inline FWCore/Services/plugins/BuildFile.xml]: @Dr15Jones , this looks good to me.

**Comment by Dr15Jones** [comment]
+1

**Comment by davidlange6** [comment]
apologies that I missed the discussion - what warnings are we trying to work around in this PR? 
(ok, I see I have some invalid tool warnings to clean up...but thats not the issue)

**Comment by fwyzard** [comment]
Boost has some "maybe-uninitialized" variables, and since we normally compile with `-Werror=maybe-uninitialized` those make the compilation fail.
My workaround was to add `-Wno-maybe-uninitialized` to the BuildFile, and Shahzad asked to split the plugin so that the flag could be applied only to the new service.

**Comment by davidlange6** [comment]
ok , thanks - hopefully a new version of boost fixes such issues:)

**Comment by davidlange6** [comment]
+1

**Pull request merged**